ext/gmp/gmp.c

```diff
@@ -1803,15 +1803,21 @@ ZEND_FUNCTION(gmp_random_bits)
 		RETURN_THROWS();
 	}
 
-	if (bits <= 0) {
-		zend_argument_value_error(1, "must be greater than or equal to 1");
+#if SIZEOF_SIZE_T == 4
+	const zend_long maxbits = ULONG_MAX / GMP_NUMB_BITS;
+#else
+	const zend_long maxbits = INT_MAX;
```

Comment:
This still can overflow. libgmp does this check, but afterwards calculates ((new_alloc) * sizeof (mp_limb_t)), where `sizeof(mp_limb_t)` is certainly larger than 2 (I assume it's usually four or eight).

Reply:
Disregard. This is fine; I was confused by 32bit using ULONG_MAX, but 64bit using INT_MAX. The 32bit case looks fishy (for LP64), though.